### ISSUE TITLE
Always display type of block in administration interface

### ISSUE THREAD

**Issue by DavidFeller** [sulu/sulu#5836]
# Problem description

The attached screenshot shows a snipped we use for a mega menu. It makes extensive use of blocks in blocks. At the moment the elements’ names are only shown if a block/element is collapsed. This is great, but sometimes it would make sense to also show this information while editing an element to keep track of the layer you are currently working on. If you are 3 or 4 layers down it becomes quite hard to know “where you are”.

![Bildschirmfoto_2021-02-23_um_10_10_41](https://user-images.githubusercontent.com/2094727/108827316-c2138b80-75c5-11eb-95c0-2d25e371b94b.jpg)

# Proposed solution

Always show the element's type and/or show some kind of indication on which layer the user is currently working (perhaps as a number or a different background color per layer).

**Comment by luca-rath** [comment]
Just a few thoughts about this ...

I agree, that it would make sense to always show the block type, but if there is just one block type, there should be something like a headline or label and not a select.

About the layer indicator, I don't think that's necessary, because the indentation is already a layer indicator.
Especially I would not make different backgrounds for the layers.

**Comment by alexander-schranz** [comment]
Yeah I agree here with @luca-rath it should not be a select to display the type here. Just a grey label like its in the closed state. This makes also the developer in control to display nothing when they want it by not given the type any meta title.

**Comment by luca-rath** [comment]
I like the idea to control the visibility of the label with the meta title. That also has to be taken into account for the collapsed state.

**Comment by alexander-schranz** [comment]
Besides the issue that it make sense to display the type title always when its set. In this case of @DavidFeller I would recommend to give also the block a meta title I think this would improve the UX also for the content manager at current state. e.g.:

```xml
<block name="columns" default-type="column" minOccurs="0">
    <meta>
        <title lang="de">Spalten</title>
        <title lang="en">Columns</title>
    </meta>

    <!-- ... -->
```

**Comment by DavidFeller** [comment]
I agree with both of a you. A label is absolutely sufficient. I just wanted to throw in some ideas :)

**Comment by alexander-schranz** [comment]
Additional you can also improve the UX for the content manager to define a better text for the add button instead of Add block:

```xml
<block name="columns" default-type="column">
    <meta>
        <title lang="de">Spalten</title>
        <title lang="en">Columns</title>
    </meta>

    <params>
        <param name="add_button_text">
            <meta>
                <title lang="en">Add column</title>
                <title lang="de">Spalte hinzufügen</title>
            </meta>
        </param>
    </params>
```

Requires I think Sulu 2.2 (https://docs.sulu.io/en/2.2/reference/content-types/block.html).

**Comment by DavidFeller** [comment]
@alexander-schranz: Oh cool! I wasn't aware of that. Great feature!
